pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -443,13 +330,171 @@ func (r *ReconcileComplianceScan) phaseAggregatingHandler(instance *complianceop
 }
 
 func (r *ReconcileComplianceScan) phaseDoneHandler(instance *complianceoperatorv1alpha1.ComplianceScan, logger logr.Logger) (reconcile.Result, error) {
+	var nodes corev1.NodeList
+	var err error
 	logger.Info("Phase: Done", "ComplianceScan scan", instance.ObjectMeta.Name)
+	if nodes, err = getTargetNodes(r, instance); err != nil {
+		log.Error(err, "Cannot get nodes")
+		return reconcile.Result{}, err
+	}
+
+	if err := r.deleteScanPods(instance, nodes, logger); err != nil {
+		log.Error(err, "Cannot delete scan pods")
+		return reconcile.Result{}, err
+	}
+
 	if err := r.deleteResultServer(instance, logger); err != nil {
+		log.Error(err, "Cannot delete result server")
 		return reconcile.Result{}, err
 	}
+
+	if err = r.deleteRootCASecret(instance, logger); err != nil {
+		log.Error(err, "Cannot delete CA secret")
+		return reconcile.Result{}, err
+	}
+
+	if err = r.deleteResultServerSecret(instance, logger); err != nil {
+		log.Error(err, "Cannot delete result server cert secret")
+		return reconcile.Result{}, err
+	}
+
+	if err = r.deleteResultClientSecret(instance, logger); err != nil {
+		log.Error(err, "Cannot delete result client cert secret")
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileComplianceScan) handleRootCASecret(instance *complianceoperatorv1alpha1.ComplianceScan, logger logr.Logger) error {
+	exist, err := secretExists(r.client, RootCAPrefix+instance.Name, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+	if exist {
+		return nil
+	}
+
+	logger.Info("creating CA", "instance", instance.Name)
+	secret, err := makeCASecret(instance, common.GetComplianceOperatorNamespace())
+	if err != nil {
+		return err
+	}
+
+	// Create the CA secret.
+	err = r.client.Create(context.TODO(), secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	if err = controllerutil.SetControllerReference(instance, secret, r.scheme); err != nil {
```

Comment:
If we're deleting them should we still set ownership?

Reply:
We shouldn't. But this function creates it.